File: mvvmwithoutcachingjavadagger/src/main/java/com/mfcwl/dagger/di/ViewModelFactoryModule.java
package com.mfcwl.dagger.di;

import androidx.lifecycle.ViewModelProvider;

import com.mfcwl.dagger.viewmodels.ViewModelProviderFactory;

import dagger.Binds;
import dagger.Module;

@Module
public abstract class ViewModelFactoryModule {

    @Binds
    public abstract ViewModelProvider.Factory bindViewModelFactory(ViewModelProviderFactory modelProviderFactory);
}

File: powerfulandroidapps_processdeathissuefix/src/main/java/com/mfcwl/library/powerfulandroidapps_processdeathissuefix/di/ViewModelFactoryModule.kt
package com.mfcwl.library.powerfulandroidapps_processdeathissuefix.di

import androidx.lifecycle.ViewModelProvider
import com.mfcwl.library.powerfulandroidapps_processdeathissuefix.viewmodels.ViewModelProviderFactory
import dagger.Binds
import dagger.Module

@Module
abstract class ViewModelFactoryModule {

    @Binds
    abstract fun bindViewModelFactory(factory: ViewModelProviderFactory): ViewModelProvider.Factory
}